### DISCUSSION TITLE
IC ticket ideas.

### DISCUSSION THREAD

**Discussion by Scwestbrook** [ChAoSUnItY/EkiLib#9] [Ideas]
What if you were to make the IC ticket work for a set amount of rides? Like you would have an IC ticket vendor, but instead of the number doing nothiing, you would put in the number of rides you wanted it to work for, and it would double it (For exiting)

**Comment by ChAoSUnItY** [comment]
Sorry for lazy updating, but at current state, IC ticket is designed to store mutable amount of money, not mutable amount of rides, although this thing is also exists in real world, but I think adding another item to have this feature is much better.

For IC ticket example, you may want to take a look at [Suica Card](https://en.wikipedia.org/wiki/Suica) or [Easy Card](https://en.wikipedia.org/wiki/EasyCard).

**Reply by Scwestbrook** [reply]
Hmmm... What if you were also to update the ticket vendor with a new config for server/client (server overrides client) like this:
<code>#What materials can be used to buy tickets
buyTicketsWith: minecraft:emerald,economyinc:item_tenb;</code>
and then, rather than you entering a number. GUI ideas:
You could put a dropdown if possible, labeled "Choose Ticket Type" with IC ticket, Normal ticket, Multiple ride ticket, or unlimited ride ticket. If you put in normal, you put currency in a slot and get one ticket. For IC you put currency in a slot and get that much value on your ticket. For Unlimited, you set how long until it expires in ticks and put in that much money. Rates can be changed in server config.

**Reply by Scwestbrook** [reply]
Do you think that redoing the vendor's GUI would be good?

**Reply by ChAoSUnItY** [reply]
Vendor's GUI is fine, but several features are testing, so I cannot tell you what will be on future update.

**Comment by Scwestbrook** [comment]
Ok, thanks. I love this mod! it's my favorite, along with the eki mod for
decoration.
